changelog.d/12304.misc

```diff
@@ -0,0 +1 @@
+Remove unused `auth_event_ids` argument plumbing.
```

Comment:
```suggestion
Refactor `create_new_client_event` to use a new parameter, `state_event_ids`, which accurately describes the usage with [MSC2716](https://github.com/matrix-org/matrix-doc/pull/2716) instead of abusing `auth_event_ids`.
```

I think we can put the two PRs into one changelog entry.